apps/prairielearn/src/tests/courseEditor.test.ts

```diff
@@ -423,33 +413,36 @@ async function getFiles(options): Promise<Set<string>> {
   });
 }
 
+let currentPage$: cheerio.CheerioAPI;
+let currentUrl: string;
 function testEdit(params) {
+  let __csrf_token: string;
```

Comment:
This is still not awesome because there is global state being messed with. The solution to this would be to make sure every object in `testEditData` has a url on it, but the downside of that is some code duplication. I'm not sure which way of doing things is better.

Reply:
I just looked into rewriting this without any global state and its messy because some of the tests follow a redirect, and then use this to pass that URL on to the next test. So I think we can just leave it like this. Having a couple of global, typed variables is still much improvement over just stuffing things into untyped `locals` as it was before.